main.go

```diff
@@ -45,15 +46,15 @@ receivers:
   otlp:
     protocols:
       http:
-        endpoint: localhost:` + strconv.Itoa(httpPortFlag) + `
+        endpoint: ` + hostFlag + `:` + strconv.Itoa(httpPortFlag) + `
       grpc:
-        endpoint: localhost:` + strconv.Itoa(grpcPortFlag) + `
+        endpoint: ` + hostFlag + `:` + strconv.Itoa(grpcPortFlag) + `
```

Comment:
This is starting to get a little unwieldy. We should probably explore using a template language in the future @CtrlSpice like https://pkg.go.dev/text/template